js/deer-render.js

```diff
@@ -310,59 +313,189 @@ DEER.TEMPLATES.lines = function (obj, options = {}) {
             }
+            const classes = Object.keys(POSITIONS)
             const unassignmentButtons = elem.querySelectorAll("i.unassign")
             for (const r of unassignmentButtons) {
-                r.addEventListener("click",e=>{
+                r.addEventListener("click", e => {
                     e.preventDefault()
+                    e.stopPropagation()
                     const forLine = e.target.closest("line")
-                    if(forLine === null) { return false }
+                    if (forLine === null) { return false }
+                    const location = Array.from(forLine.classList).filter(val => classes.includes(val))?.[0]
                     let thisLine = forLine
-                    while (thisLine) {
-                        if(!thisLine.classList.contains(location)) { break }
-                        thisLine.classList.remove("selected")
-                        thisLine.classList.remove("just")
+                    while (thisLine?.classList.contains(location)) {
+                        thisLine.classList.remove(location, "located", "selected", "just")
                         thisLine = thisLine.nextElementSibling
                     }
+                    saveBtn.style.visibility="visible"
                 })
             }
             const selected = elem.querySelectorAll(".selected")
             for (const s of selected) {
-                s.classList.add("located", assignment.split(/\s/).reduce((response,word)=> response+=word.slice(0,1),''))
+                s.classList.add("located", assignment.split(/\s/).reduce((response, word) => response += word.slice(0, 1), ''))
                 s.classList.remove("just", "selected")
             }
+            saveBtn.addEventListener("click", saveLocations)
+            function saveLocations() {
+                const locationMap = new Map()
+                allLines.forEach(line => {
+                    const location = Array.from(line.classList).filter(val => classes.includes(val))?.[0]
+                    locationMap.set(line.getAttribute("title"), location ?? false)
+                })
+
+                const locationAnnotationId = document.querySelector("div.page").getAttribute("data-marginalia")
+                if(!locationAnnotationId) {
+                    throw new Error("URI for Annotation Location could not be found.")
+                }
+
+                /**
+                 * TODO: This is a dirty trick to make this save reliably. The annotation should be targeted to 
+                 * just the lines within the Canvas and this map should be iterated through instead.
+                 */
+                const locationAnnotation = {
+                    "@id": locationAnnotationId,
+                    "@type": "Annotation",
+                    "@context": "http://www.w3.org/ns/anno.jsonld",
+                    target: c['@id'],
+                    body: { locations: Object.fromEntries(locationMap.entries()) },
```

Comment:
Could have used your standard "Set" annotation so it had an array of items, but this will work.  When having a picker in a DEER form, it will make that Set style annotation.  When I make an Annotation to connect lines to Glosses, should I do it like 

`body: {"_tpen_lines": {"@type":"Set", "items":["line/1", "line/n"]}}` 
or 
`body: {"_tpen_lines": ["line/1", "line/n"]}`

Reply:
Also, I take it you just want me to grab the numeric line number and store it as a number, not a "line/nnnnn" string.

Reply:
It should probably be an ItemList but I wanted to have a Map instead of an array, so Set wouldn't have worked either.

Reply:
The reference is just to make the connection, so it doesn't really matter. I think I went to just numbers because the logic was simpler, but anything without a `.` will work.

Reply:
Sounds good, mine is geared for the ItemList so that's what I will do.